dattri/task.py

```diff
@@ -90,6 +121,11 @@ def f(params, data):
         self.original_target_func = target_func
         self.target_func = flatten_func(self.model)(target_func)
 
+        if checkpoints_load_func is None:
+            self.checkpoints_load_func = _default_checkpoint_load_func
+        else:
+            self.checkpoints_load_func = checkpoints_load_func
+
```

Comment:
do we need to be careful here? now it seems like we are either using default or huggingface loading.
Will there be any other type of loading?

Reply:
The logic of this code block is. If `checkpoints_load_func` is not explicitly provided, we will use `_default_checkpoint_load_func`. `checkpoints_load_func` is provided by users, it can be huggingface loading or any other things.

Reply:
Oh got it. sorry for the misunderstanding